Assets/Scripts/Augment/AugmentImplementations/Solar/SolarBody.cs

```diff
@@ -50,9 +59,11 @@ private IEnumerator CoolDown()
         isCooldown = false;
     }
 
-    void Update()
+    void FixedUpdate()
     {
-        if (transform.parent.rotation.eulerAngles.y < sunDirection + sunDirectionSpan && transform.parent.rotation.eulerAngles.y > sunDirection - sunDirectionSpan)
+        Vector3 relativeRotationSolarpanel = rayCastOrigin.transform.TransformDirection(Vector3.up);
```

Comment:
SolarPanel, and the whole thing should be relativeRotationOfSolarPanel if you're this verbose.
Couldn't you just use the forward vector of the origin if it is a point you define yourself, or is it a reference to the actual solar panel plane?

Reply:
https://docs.unity3d.com/ScriptReference/Transform-up.html :)